toxencryptsave/reader.hpp

```diff
@@ -0,0 +1,34 @@
+#ifndef TOXENCRYPTSAVE_READER_H
+#define TOXENCRYPTSAVE_READER_H
+
+#include "data_description.hpp"
+#include "key.hpp"
+
+namespace toxencryptsave
+{
+
+struct Reader
+{
+    Data_Description<const uint8_t*> const ro_data;
+
+    explicit Reader(const uint8_t* const raw_encryptsave_data):
```

Comment:
Reader now has an implicit dependency  over some external raw pointer, not very safe. And where the length checks of input data should be?